lib/functions/Core.ts

```diff
@@ -45,6 +89,27 @@ export abstract class BaseFunction<Operator> {
 
   protected abstract handleInvalidTypes(args: Term[]): never;
 
+  /**
+   * Searches in a depth fist way for the best matching overload.
+   * @param args
+   * @private
+   */
+  private search(args: Term[]): E.SimpleApplication {
+    const searchStack: { node: OverloadNode; index: number }[] = [];
+    const startIndex = 0;
+    searchStack.push(...this.overloads.getSubTreeWithArg(args[startIndex]).map(node =>
```

Comment:
Why is the copy needed?

Reply:
We don't need a copy, I just need to append the whole array to the other one in place. How would you do this?

Reply:
Oh ok, I misunderstood, nvm my comment then.

(some documentation would have helped here 😅)

Reply:
You're right the function isn't that trivial when you didn't write it yourself. :)